pkg/domain/infra/abi/containers.go

```diff
@@ -294,6 +284,27 @@ func (ic *ContainerEngine) ContainerRm(ctx context.Context, namesOrIds []string,
 		names = append(names, id)
 	}
 
+	// Attempt to remove named containers directly from storage, if container is defined in libpod
+	// this will fail, if options.Storage is not specified, code will fall through to removing
+	// the container from libpod.`
+	i := 0
+	for _, ctr := range names {
+		report := entities.RmReport{Id: ctr}
+		if err := ic.Libpod.RemoveStorageContainer(ctr, options.Force); err != nil {
+			// remove container names that we successfully deleted
+			names[i] = ctr
+			i++
+			report.Err = err
+			if options.Storage {
+				reports = append(reports, &report)
+				return reports, nil
```

Comment:
This doesn't seem correct - the old `--storage` behavior was to try all images, not fail when only one failed.

Reply:
No it was not, the code, would attempt to remove the nameOrIDs from containerStorage.  But now it will do it if the container exists in storage and not in libpod.  Making the storage flag useless.  I am going to just hide it and remove it from docs.

Reply:
What I am saying is that the new code will return an error after *any single image* fails. The old code tried *all* images, even if one failed.

Reply:
I think we just remove the storage option altogether, if the image does not exists in libpod, it will report and error now, and it will be removed if it is just in container/storage.